src/frontends/onnx/frontend/src/op/layer_norm.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/layer_norm.hpp"
+#include "ngraph/node.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <numeric>
+
+#include "default_opset.hpp"
+#include "exceptions.hpp"
+#include "ngraph/validation_util.hpp"
+
+#include "ngraph/axis_set.hpp"
+#include "ngraph/op/mvn.hpp"
+#include "ngraph/opsets/opset5.hpp"
+#include "ngraph/validation_util.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+//    Decompose LayerNorm(x) to MVN(x) * gamma + beta
+OutputVector layer_norm(const Node& node) {
+    auto nodes = node.get_ng_inputs();
+    auto num_nodes = nodes.size();
+    NGRAPH_CHECK(num_nodes == 2 || num_nodes == 3,
+                 "LayerNormalization takes 2/3 inputs. Provided " + std::to_string(num_nodes));
+
+    // input
+    Output<ngraph::Node> input = nodes[0];
+
+    float eps = node.get_attribute_value<float>("epsilon", 1e-5);
+    int64_t axis = node.get_attribute_value<int64_t>("axis", 0);
+    //const std::int32_t stash_type{node.get_attribute_value<std::int32_t>("stash_type", 1)};
+    std::vector<int64_t> reduction_axes;
+    const auto input_shape_size = input.get_partial_shape().rank().get_max_length();
```

Comment:
This will not work correctly for a dynamic (unknown) rank case. It will generate a huge number which is later used to create a range of values by filling the `reduction_axes` vector.